core/src/main/scala/pureconfig/DerivedConverters.scala

```diff
@@ -0,0 +1,268 @@
+      val tailCo = if (hint.allowUnknownKeys) co else co.withoutKey(keyStr)
+      val tailResult = tConfigConvert.value.fromWithDefault(tailCo, default.tail)
+      combineResults(headResult, tailResult)((head, tail) => field[K](head) :: tail)
+    }
+
+    override def to(t: FieldType[K, V] :: T): ConfigValue = {
+      val keyStr = hint.configKey(key.value.toString().tail)
+      val rem = tConfigConvert.value.to(t.tail)
+      // TODO check that all keys are unique
+      vFieldConvert.value match {
+        case f: OptionConfigConvert[_] =>
+          f.toOption(t.head) match {
+            case Some(v) =>
+              rem.asInstanceOf[ConfigObject].withValue(keyStr, v)
+            case None =>
+              rem
+          }
+        case f =>
+          val fieldEntry = f.to(t.head)
+          rem.asInstanceOf[ConfigObject].withValue(keyStr, fieldEntry)
+      }
+    }
+  }
+
+  implicit final def cNilConfigConvert[Wrapped]: WrappedConfigConvert[Wrapped, CNil] = new WrappedConfigConvert[Wrapped, CNil] {
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, CNil] =
+      fail(NoValidCoproductChoiceFound(config, ConfigValueLocation(config), None))
+
+    override def to(t: CNil): ConfigValue = ConfigFactory.parseMap(Map().asJava).root()
+  }
+
+  implicit final def coproductConfigConvert[Wrapped, Name <: Symbol, V, T <: Coproduct](
+    implicit
+    coproductHint: CoproductHint[Wrapped],
+    vName: Witness.Aux[Name],
+    vFieldConvert: Lazy[ConfigConvert[V]],
+    tConfigConvert: Lazy[WrappedConfigConvert[Wrapped, T]]): WrappedConfigConvert[Wrapped, FieldType[Name, V] :+: T] =
+    new WrappedConfigConvert[Wrapped, FieldType[Name, V] :+: T] {
+
+      override def from(config: ConfigValue): Either[ConfigReaderFailures, FieldType[Name, V] :+: T] =
+        coproductHint.from(config, vName.value.name) match {
+          case Right(Some(hintConfig)) =>
+            vFieldConvert.value.from(hintConfig) match {
+              case Left(_) if coproductHint.tryNextOnFail(vName.value.name) =>
+                tConfigConvert.value.from(config).right.map(s => Inr(s))
+
+              case vTry => vTry.right.map(v => Inl(field[Name](v)))
+            }
+
+          case Right(None) => tConfigConvert.value.from(config).right.map(s => Inr(s))
+          case l: Left[_, _] => l.asInstanceOf[Either[ConfigReaderFailures, FieldType[Name, V] :+: T]]
+        }
+
+      override def to(t: FieldType[Name, V] :+: T): ConfigValue = t match {
+        case Inl(l) =>
+          // Writing a coproduct to a config can fail. Is it worth it to make `to` return a `Try`?
+          coproductHint.to(vFieldConvert.value.to(l), vName.value.name) match {
+            case Left(failures) => throw new ConfigReaderException[FieldType[Name, V] :+: T](failures)
+            case Right(r) => r
+          }
+
+        case Inr(r) =>
+          tConfigConvert.value.to(r)
+      }
+    }
+
+  // For Option[T] we use a special config converter
```

Comment:
(I know you didn't add this comment. I'm just realizing it's here and it annoys me. Feel free to ignore me.)
Please expand the comment to indicate why or remove it. The comment isn't telling me anything the name doesn't.